pangeo_forge_runner/feedstock.py

```diff
@@ -86,6 +89,44 @@ def parse_recipes(self):
 
         return recipes
 
+    @contextmanager
+    def generate_setup_py(self):
+        """
+        Auto-generate a setup.py file for use with apache beam.
+
+        Beam sends all the user code we need to workers by creating an
+        sdist off a python package. However, our feedstocks only have a
+        few python files (at best) - mostly just one (recipe.py). We do not
+        want to impose creating a setup.py file manually for all our users,
+        so instead we autogenerate one here.
+        """
+        file = dedent(
+            """
+        import setuptools
```

Comment:
I would suggest moving this template out into its own file.

This generation also needs to be watched closely for code injection based attacks.